tripy/docs/pre0_user_guides/02-compiler.md

```diff
@@ -0,0 +1,191 @@
+# Using the Compiler
+
+```{contents} Table of Contents
+:depth: 3
+```
+
+## Eager Mode vs. Compile Mode
+
+Tripy offers two modes of operation: eager mode and compile mode:
+
+### Eager Mode
+
+Eager mode is designed for development and debugging purposes:
+
+1. **Deferred Compilation**:  In eager mode, compilation is deferred until the first use of a tensor, allowing for easier debugging.
```

Comment:
Should we specify that "Eager mode" performs Lazy evaluation i.e. "Deferred compilation" and an execution?
We use "Lazy evaluation" in a couple of docs/comments so it would be better to define it formally and link it with "Eager mode".

Reply:
I removed this section from the guide.